docs/docs/hive.md

```diff
@@ -431,12 +466,120 @@ ALTER TABLE t SET TBLPROPERTIES ('storage_handler'='org.apache.iceberg.mr.hive.H
 During the migration the data files are not changed, only the appropriate Iceberg metadata files are created.
 After the migration, handle the table as a normal Iceberg table.
 
+#### Drop partitions
```

Comment:
I think it will confuse the new Hive3.x users. 

We might have to clearly mention that it works only for Hive4.x for each of these new commands addition. 

Or maybe we should have 2 docs, one for hive3.x and one for hive4.x since the difference is huge.

Reply:
@ajantha-bhat I think the Hive docs should reflect the latest documentation w.r.t to the component. The documentation already has a list of features introduced in a corresponding release. Users will figure out if the feature is present or not.

I wonder how Spark docs and other documentations manage to tackle this, given that they have features released in different releases and I don't see much differentiation among the same.

Reply:
I tend to agree with @SourabhBadhya, that the documentation should focus on the latest release of Hive.

Maybe we should also remove beta/alpha releases, as I would expect that the users will use a stable release instead. 

We have a starting paragraph about `Feature support`, which helps to understand the difference.

Do you think @ajantha-bhat that this would be enough?

Reply:
Maybe we can have a feature support matrix with hive3.x and hive4.x like we had for spark 
https://github.com/apache/iceberg/blob/apache-iceberg-1.2.0/docs/spark-writes.md#spark-writes

I agree that we can remove beta/alpha releases.